tests/functional/mesh/reload_test.go

```diff
@@ -0,0 +1,166 @@
+package mesh
```

Comment:
do we need reload_test.go? If we can keep the testing to just work_test.go that would be ideal

Reply:
I was originally thinking that we can have two different test files : 

`reload_test.go` would just test the reload functionality. 
`work_test.go` would test the work streaming process in case of a reload.

Reply:
I have put all the test in ```work_test.go```